### PR TITLE
[14.0][MIG] hr_recruitment_notification: Migration to 14.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 14.0.

Please @pedrobaeza and @joao-p-marques can you review it?

@Tecnativa TT32342

### CODE REVIEW

File: hr_recruitment_notification/tests/test_hr_application_notification.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ def test_hr_application_notification(self):
             {"name": "Test Job Application for Notification", "job_id": self.job.id}
         )
         new_application_message = application.message_ids.filtered(
-            lambda m: self.new_application_mt.description in m.body
+            lambda m: m.subtype_id.description == self.new_application_mt.description
```
Comment (pedrobaeza) [inline hr_recruitment_notification/tests/test_hr_application_notification.py]:
```suggestion
            lambda m: m.subtype_id == self.new_application_mt
```

**Review by pedrobaeza — COMMENTED** [review]
Please add in the roadmap that this module is not needed in 15.0

File: hr_recruitment_notification/tests/test_hr_application_notification.py (hunk 1 above)
Comment (victoralmau) [reply hr_recruitment_notification/tests/test_hr_application_notification.py]: If I'm not mistaken this wouldn't work, odoo defines the subtype of the `parent_id` (`hr_recruitment.mt_applicant_ne`) in the message (in v13 and in v14).

Comment (pedrobaeza) [reply hr_recruitment_notification/tests/test_hr_application_notification.py]: Uhm, I see, then what about `m.subtype_id.parent_id == self.new_applicant_mt`?

**Review by pedrobaeza — APPROVED** [review]

**Review by joao-p-marques — APPROVED** [review]

**Comment by joao-p-marques** [comment]
@pedrobaeza AFAICS I should be able to merge as [i am a declared maintainer in the manifest](https://github.com/OCA/hr/pull/1034/files#diff-aa6e218a4790c12374cbc938e646f139834656f21ad2f81fe9f0bb2849eedf1eR12) Is this normal or is it a bug in OCA bot?

**Comment by pedrobaeza** [comment]
> @pedrobaeza AFAICS I should be able to merge as [i am a declared maintainer in the manifest](https://github.com/OCA/hr/pull/1034/files#diff-aa6e218a4790c12374cbc938e646f139834656f21ad2f81fe9f0bb2849eedf1eR12) Is this normal or is it a bug in OCA bot?

The problem is that the module hasn't arrived yet to the repository, so it can't be checked that you are the real maintainer, or you want to hack it through this PR. This has a difficult solution, and not sure if @sbidoul has anything in mind.

Meanwhile, I merge:

/ocabot merge nobump

**Comment by sbidoul** [comment]
@joao-p-marques @pedrobaeza it's the mechanism to let PSC approve new maintainers.

**Comment by joao-p-marques** [comment]
Ok... As I was a maintainer for the module in the previous version, so I assumed it would work... Maybe the bot could check the maintainers defined in the previous branch if it is a migration PR?

**Comment by sbidoul** [comment]
> Ok... As I was a maintainer for the module in the previous version, so I assumed it would work... Maybe the bot could check the maintainers defined in the previous branch if it is a migration PR?

It could yes :) There is even a tracking issue for that: https://github.com/OCA/oca-github-bot/issues/122

**Pull request merged**